Products/ZenUtils/ShardedBTree.py

```diff
@@ -0,0 +1,150 @@
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2017, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+
+from BTrees.OOBTree import OOBTree
+from OFS.SimpleItem import SimpleItem
+from persistent.list import PersistentList
+from itertools import chain
+from zlib import adler32
+
+
+DEFAULT_N_SHARDS = 127
+
+
+def default_hash_func(data):
+    """
+    adler32 chosen as hashing mechanism since we dont need a cryptographic hashing
+    """
+    # In python < 3.0 adler32 returns a signed value
+    return adler32(data) & 0xffffffff
+
+
+class ShardedBTree(SimpleItem):
+    """
+    Sharded BTree to reduce the number of ConflictErrors
+    """
+    def __init__(self, n_shards=DEFAULT_N_SHARDS, hash_func=default_hash_func):
+        """ """
+        super(ShardedBTree, self).__init__()
+        self.n_shards = n_shards
+        self.hash_func = hash_func
+        self.shards = self._build_shards(self.n_shards)
+
+    def _build_shards(self, n_shards):
+        shards = PersistentList()
```

Comment:
I bet you get a speed increase if you make this a simple list instead of a PersistentList. Worth a try, anyway.